src/overlay/Peer.cpp

```diff
@@ -382,23 +383,27 @@ Peer::sendPeers()
     newMsg.type(PEERS);
     uint32 maxPeerCount = std::min<uint32>(50, newMsg.peers().max_size());
 
+    auto keep = [&](PeerRecord pr) {
+        return !pr.getAddress().isPrivate() && pr.getAddress() != mAddress;
+    };
+
     // send top peers we know about
-    vector<PeerRecord> peerList;
-    PeerRecord::loadPeerRecords(mApp.getDatabase(), 50, mApp.getClock().now(),
-                                [&](PeerRecord const& pr) {
-                                    bool r = peerList.size() < maxPeerCount;
-                                    if (r)
-                                    {
-                                        if (!pr.getAddress().isPrivate() &&
-                                            pr.getAddress() != mAddress)
-                                        {
-                                            peerList.emplace_back(pr);
-                                        }
-                                    }
-                                    return r;
-                                });
-    newMsg.peers().reserve(peerList.size());
-    for (auto const& pr : peerList)
+    auto peers = mApp.getOverlayManager().getRandomPeerSource().getRandomPeers(
```

Comment:
just realized that there is a small bug here (that was there in master as well I think): because we call `backoff` when we connect to peers, that means that pending peers (that may be good!) are not returned in this list.

There is another issue here (that may allow us to fix that issue): we are sharing the "randomPeerSource" with the one used for connecting to other peers.
We should not do this: this creates a bunch of problems as it makes the distribution potentially biases for connecting outbound.
By separating those, we can also change the select used for building this list of peers: we don't need to filter by date at all for this scenario, only failure count (don't use peers with failure count more than some high number like 9).